OsmAnd/src/net/osmand/plus/activities/actions/ShareLocation.java

```diff
@@ -44,16 +44,16 @@ public void onClick(DialogInterface dialog, int which) {
 				final int zoom = args.getInt(MapActivityActions.KEY_ZOOM);
 				try {
 					final String shortOsmUrl = MapUtils.buildShortOsmUrl(latitude, longitude, zoom);
-					final String appLink = "http://download.osmand.net/go?lat=" + ((float) latitude) + "&lon=" + ((float) longitude) + "&z=" + zoom;
-					String sms = mapActivity.getString(R.string.send_location_sms_pattern, shortOsmUrl, appLink);
+					final String geoUrl = MapUtils.buildGeoUrl(latitude, longitude, zoom);
+					String sms = mapActivity.getString(R.string.send_location_sms_pattern, geoUrl, shortOsmUrl);
 					if (which == 0) {
-						sendEmail(shortOsmUrl, appLink);
```

Comment:
This is actually should stay app link because we can't rely that receiver will have OsmAnd install instead we rely on proper internet address